awx/ui_next/src/screens/ObservabilityMetrics/LineChart.jsx

```diff
@@ -0,0 +1,272 @@
+      .text(helpText);
+
+    const svg = d3
+      .select('#chart')
+      .append('svg')
+      .attr('width', `${width + margin}px`)
+      .attr('height', `${height + margin}px`)
+      .append('g')
+      .attr('transform', `translate(${margin}, ${margin})`);
+
+    /* Add line into SVG */
+    const line = d3
+      .line()
+      .curve(d3.curveMonotoneX)
+      .x(d => xScale(d.x))
+      .y(d => yScale(d.y));
+
+    const lines = svg.append('g');
+
+    lines
+      .selectAll('.line-group')
+      .data(data)
+      .enter()
+      .append('g')
+      .attr('class', 'line-group')
+      .append('path')
+      .attr('class', 'line')
+      .style('fill', 'none')
+      .attr('d', d => line(d.values))
+      .style('stroke', (d, i) => color(i))
+      .style('stroke-width', '3px');
+
+    /* Add circles in the line */
+    lines
+      .selectAll('circle-group')
+      .data(data)
+      .enter()
+      .append('g')
+      .style('fill', (d, i) => color(i))
+      .selectAll('circle')
+      .data(d => d.values)
+      .enter()
+      .append('g')
+      .attr('class', 'circle')
+      .on('mouseover', (d, i) => {
+        if (data.length) {
+          renderTooltip(d, i);
+        }
+      })
+      .on('mouseout', () => {
+        removeTooltip();
+      })
+      .append('circle')
+      .attr('cx', d => xScale(d.x))
+      .attr('cy', d => yScale(d.y))
+      .attr('r', circleRadius)
+      .on('mouseover', () => {
+        d3.select(this)
+          .transition()
+          .duration(duration)
+          .attr('r', circleRadiusHover);
+      })
+      .on('mouseout', () => {
+        d3.select(this)
+          .transition()
+          .duration(duration)
+          .attr('r', circleRadius);
+      });
+
+    /* Add Axis into SVG */
+    const xAxis = d3
+      .axisBottom(xScale)
+      .ticks(data[0].values.length > 5 ? data[0].values.length : 5);
+    const yAxis = d3.axisLeft(yScale).ticks(5);
+
+    svg
+      .append('g')
+      .attr('class', 'x axis')
+      .attr('transform', `translate(0, ${height - margin})`)
+      .call(xAxis)
+      .append('text')
+      .attr('x', 30)
+      .attr('y', 30)
+      .attr('fill', '#000')
+      .text(i18n._(t`Count`));
+
+    svg
+      .append('g')
+      .attr('class', 'y axis')
+      .call(yAxis)
+      .append('text')
+      .attr('y', -30)
+      .attr('x', 0)
+      .attr('transform', 'rotate(-90)')
+      .attr('fill', '#000')
+      .text(i18n._(t`Values`));
+  }, [data, i18n, helpText]);
+
+  useEffect(() => {
+    draw();
+  }, [count, draw]);
+
+  useEffect(() => {
+    function handleResize() {
+      draw();
+    }
+
+    window.addEventListener('resize', handleResize);
+
+    handleResize();
+
+    return () => window.removeEventListener('resize', handleResize);
+  }, [draw]);
+
+  return <div id="chart" />;
+}
+
+const withPageContext = Component => {
+  return function contextComponent(props) {
+    return (
+      <PageContextConsumer>
```

Comment:
What is `PageContextConsumer` for? 

(Just to see what would happen, I removed it in the code and as far as I can tell everything still works just fine without it?)